modules/reindex/src/internalClusterTest/java/org/elasticsearch/index/reindex/ReindexPluginMetricsIT.java

```diff
@@ -0,0 +1,263 @@
+        );
+        assertHitCount(prepareSearch("source").setSize(0), 4);
+
+        final String dataNodeName = internalCluster().getRandomNodeName();
+        final TestTelemetryPlugin testTelemetryPlugin = internalCluster().getInstance(PluginsService.class, dataNodeName)
+            .filterPlugins(TestTelemetryPlugin.class)
+            .findFirst()
+            .orElseThrow();
+
+        // Copy all the docs
+        ReindexRequestBuilder copy = reindex().source("source").destination("dest").refresh(true);
+        assertThat(copy.get(), matcher().created(4));
+        assertHitCount(prepareSearch("dest").setSize(0), 4);
+
+        // Use assertBusy to wait for all threads to complete so we get deterministic results
+        assertBusy(() -> {
+            testTelemetryPlugin.collect();
+            List<Measurement> measurements = testTelemetryPlugin.getLongHistogramMeasurement(REINDEX_TIME_HISTOGRAM);
+            assertThat(measurements.size(), equalTo(1));
+        });
+
+        // Now none of them
+        createIndex("none");
+        copy = reindex().source("source").destination("none").filter(termQuery("foo", "no_match")).refresh(true);
+        assertThat(copy.get(), matcher().created(0));
+        assertHitCount(prepareSearch("none").setSize(0), 0);
+        // wait for all threads to complete so that we get deterministic results
+        // waitUntil(() -> (tps[0] = tp.stats()).stats().stream().allMatch(s -> s.active() == 0));
+
+        // testTelemetryPlugin.collect();
+        // measurements = testTelemetryPlugin.getLongHistogramMeasurement(REINDEX_TIME_HISTOGRAM);
+        assertBusy(() -> {
+            testTelemetryPlugin.collect();
+            List<Measurement> measurements = testTelemetryPlugin.getLongHistogramMeasurement(REINDEX_TIME_HISTOGRAM);
+            assertThat(measurements.size(), equalTo(2));
+        });
+
+        // Now half of them
+        copy = reindex().source("source").destination("dest_half").filter(termQuery("foo", "a")).refresh(true);
+        assertThat(copy.get(), matcher().created(2));
+        assertHitCount(prepareSearch("dest_half").setSize(0), 2);
+
+        // wait for all threads to complete so that we get deterministic results
+        // waitUntil(() -> (tps[0] = tp.stats()).stats().stream().allMatch(s -> s.active() == 0));
+
+        // testTelemetryPlugin.collect();
+        // measurements = testTelemetryPlugin.getLongHistogramMeasurement(REINDEX_TIME_HISTOGRAM);
+        assertBusy(() -> {
+            testTelemetryPlugin.collect();
+            List<Measurement> measurements = testTelemetryPlugin.getLongHistogramMeasurement(REINDEX_TIME_HISTOGRAM);
+            assertThat(measurements.size(), equalTo(3));
+        });
+
+        // Limit with maxDocs
+        copy = reindex().source("source").destination("dest_size_one").maxDocs(1).refresh(true);
+        assertThat(copy.get(), matcher().created(1));
+        assertHitCount(prepareSearch("dest_size_one").setSize(0), 1);
```

Comment:
Since this tests are meant to test that we do collect the metrics on the different reindex operations I wonder if we should assert that reindex and related operations work as expected, maybe we can simplify this a bit and remove these assertions? wdyt?

Reply:
> Do we want to remove them so the test would run faster or because it really doesn't matter if the operation succeeds or not - I guess metrics should get collected even for failed operations, right ?

Yes, that's my thinking, since we're just interested in testing the metric collection in these tests.

Reply:
> > Do we want to remove them so the test would run faster or because it really doesn't matter if the operation succeeds or not - I guess metrics should get collected even for failed operations, right ?
> 
> Yes, that's my thinking, since we're just interested in testing the metric collection in these tests.

Sure let me remove it. I can also remove the refresh(true) option in that case.